packages/test/test-end-to-end-tests/src/test/rootDatastores.spec.ts

```diff
@@ -420,5 +420,44 @@ describeNoCompat("Named root data stores", (getTestObjectProvider) => {
             assert.equal(aliasResult3, "Conflict");
             assert.ok(await getRootDataStore(dataObject3, alias));
         });
+
+        itExpects("Aliasing datastores summarized before the alias op is sent and after the attach op is sent " +
+        "does not cause a datastore corruption issue, and remote aliased datastores are referenced", [], async () => {
+            await setupContainers({
+                ... testContainerConfig,
+                runtimeOptions: {
+                    summaryOptions: {
+                        disableSummaries: true,
+                    },
+                    gcOptions: {
+                        gcAllowed: true,
+                    },
+                },
+            });
+
+            const containerRuntime1 = runtimeOf(dataObject1);
+            const aliasableDataStore1 = await containerRuntime1.createDataStore(packageName);
+    
+            (aliasableDataStore1 as any).fluidDataStoreChannel.bindToContext();
```

Comment:
Can you add a comment explaining that this is casting to any but its because we are trying to repro a race condition.